vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1499,6 +1521,26 @@ func resourceVSphereComputeClusterApplyVsanConfig(d *schema.ResourceData, meta i
 		return fmt.Errorf("cannot apply vsan service on cluster '%s': %s", d.Get("name").(string), err)
 	}
 
+	if !d.Get("vsan_esa_enabled").(bool) {
```

Comment:
Found a bug here, we separate dedup and compression in another reconfigure call, due to these two fields are conflict with vSAN ESA, they cannot be set in same spec along with vsan esa, but if we only check whether vsan esa is false, the reconfigure will also be called when we try to disable vsan esa, it's definitely wrong. The correct check is only vsan is true **AND** vsan esa is false, these two fields could be added to conf.